kasa/tests/test_sslaestransport.py

```diff
@@ -0,0 +1,369 @@
+    host = "127.0.0.1"
+    port_override = 12345
+    config = DeviceConfig(
+        host, credentials=Credentials("foo", "bar"), port_override=port_override
+    )
+    transport = SslAesTransport(config=config)
+
+    assert str(transport._app_url) == f"https://127.0.0.1:{port_override}"
+
+
+class MockSslAesDevice:
+    BAD_USER_RESP = {
+        "error_code": SmartErrorCode.SESSION_EXPIRED.value,
+        "result": {
+            "data": {
+                "code": -60502,
+            }
+        },
+    }
+
+    BAD_PWD_RESP = {
+        "error_code": SmartErrorCode.INVALID_NONCE.value,
+        "result": {
+            "data": {
+                "code": SmartErrorCode.SESSION_EXPIRED.value,
+                "encrypt_type": ["3"],
+                "key": "Someb64keyWithUnknownPurpose",
+                "nonce": "1234567890ABCDEF",  # Whatever the original nonce was
+                "device_confirm": "",
+            }
+        },
+    }
+
+    class _mock_response:
+        def __init__(self, status, request: dict):
+            self.status = status
+            self._json = request
+
+        async def __aenter__(self):
+            return self
+
+        async def __aexit__(self, exc_t, exc_v, exc_tb):
+            pass
+
+        async def read(self):
+            if isinstance(self._json, dict):
+                return json_dumps(self._json).encode()
+            return self._json
+
+    def __init__(
+        self,
+        host,
+        *,
+        status_code=200,
+        want_default_username: bool = False,
+        do_not_encrypt_response=False,
+        send_response=None,
+        sequential_request_delay=0,
+        send_error_code=0,
+        secure_passthrough_error_code=0,
+        digest_password_fail=False,
+    ):
+        self.host = host
+        self.http_client = HttpClient(DeviceConfig(self.host))
+        self.encryption_session: AesEncyptionSession | None = None
+        self.server_nonce = secrets.token_bytes(8).hex().upper()
+        self.handshake1_complete = False
+
+        # test behaviour attributes
+        self.status_code = status_code
+        self.send_error_code = send_error_code
+        self.secure_passthrough_error_code = secure_passthrough_error_code
+        self.do_not_encrypt_response = do_not_encrypt_response
+        self.want_default_username = want_default_username
+        self.digest_password_fail = digest_password_fail
+
+    async def post(self, url: URL, params=None, json=None, data=None, *_, **__):
+        if data:
+            json = json_loads(data)
+        res = await self._post(url, json)
+        return res
+
+    async def _post(self, url: URL, json: dict[str, Any]):
+        method = json["method"]
+
+        if method == "login" and not self.handshake1_complete:
+            return await self._return_handshake1_response(url, json)
+        if method == "login" and self.handshake1_complete:
+            return await self._return_handshake2_response(url, json)
```

Comment:
```suggestion
        if method == "login" and not self.handshake1_complete:
            return await self._return_handshake1_response(url, json)

        if method == "login" and self.handshake1_complete:
            return await self._return_handshake2_response(url, json)
```
These belong to different if-elif-else statements.